src/onemod/pipeline.py

```diff
@@ -40,8 +41,42 @@ class Pipeline(BaseModel):
     directory: Path
     config: Config = Config()
     groupby_data: Path | None = None
+    _module: Path | None = None
     _stages: dict[str, Stage] = {}
 
+    def __init__(
+        self,
+        module: Path | str | None = None,
+        stages: list[Stage] | None = None,
+        **kwargs,
+    ) -> None:
+        """Create pipeline instance."""
+        super().__init__(**kwargs)
+        self.set_module(module)
+        if stages is not None:
+            self.add_stages(stages)
+
+    @computed_property
+    def type(self) -> str:
```

Comment:
Should we change `type` to `class_name` as well?

Reply:
I prefer keeping `type` here